src/libstore/include/nix/store/build/build-log.hh

```diff
@@ -0,0 +1,84 @@
+#pragma once
+///@file
+
+#include "nix/util/logging.hh"
+#include "nix/util/serialise.hh"
+
+#include <list>
+#include <map>
+#include <string>
+
+namespace nix {
+
+/**
+ * Line buffering and log tracking for build output.
+ *
+ * This class handles:
+ * - Owning the build Activity for logging
```

Comment:
I would expect the build Activity to be broader than just the `builder` child process which this class seems to focus on (see eg doc for `operator()`). We also have things like pre and post build hooks, or the built-in reference checks that are part of the broader process of building.